src/Blocks/BaseBlock.php

```diff
@@ -120,4 +112,136 @@ public static function render_frontend(array $attributes): string
         return '<div data-render="' . self::get_full_block_name() .
             '" data-attributes="' . htmlspecialchars($json) . '"></div>';
     }
+
+    /**
+     * Register scripts and styles for a block
+     */
+    public static function enqueue_editor_assets(): void
+    {
+        static::enqueue_editor_script();
+        static::enqueue_frontend_style();
+        static::enqueue_editor_style();
+    }
+
+    /**
+     * Enqueue assets for the frontend IF the blocks is present.
+     */
+    public static function enqueue_frontend_assets(): void
+    {
+        $full_name = static::get_full_block_name();
+        $beta_blocks = [];
+
+        $to_look_for = $full_name;
+        if (in_array($full_name, $beta_blocks, true)) {
+            $to_look_for .= '-beta';
+        }
+
+        if (! is_preview() && ! BlockList::has_block($to_look_for)) {
+            return;
+        }
+
+        static::enqueue_frontend_script();
+        static::enqueue_frontend_style();
+    }
+
+    /**
+     * Editor script
+     */
+    public static function enqueue_editor_script(): void
+    {
+        $filepath = static::get_dir_path() . 'EditorScript.js';
+        if (! file_exists($filepath)) {
+            return;
+        }
+
+        wp_enqueue_script(
+            static::get_full_block_name() . '-editor-script',
+            static::get_url_path() . 'EditorScript.js',
+            'planet4-blocks-editor-script',
+            \P4GBKS\Loader::file_ver($filepath),
```

Comment:
Hi @lithrel , here we can use a master-theme file_ver function (i.e [theme_file_var()](https://github.com/greenpeace/planet4-master-theme/blob/main/src/Loader.php#L190)), right?

Reply:
They don't take the same argument (one takes the full filepath, the other a path relative to the theme), so we need to adapt one or the other; but after that it's the same `filectime` principle yes

Reply:
I updated the PR, please have a look.